Explorer/Assets/DCL/Ipfs/IpfsRealm.cs

```diff
@@ -0,0 +1,127 @@
+using System.Text;
+using System.Threading;
+using UnityEngine;
+
+namespace Ipfs
+{
+    public class IpfsRealm : IIpfsRealm, IEquatable<IpfsRealm>
+    {
+        private readonly IWeb3IdentityCache web3IdentityCache;
+        private readonly IWebRequestController webRequestController;
+        private readonly List<string> sceneUrns;
+        private readonly URLBuilder urlBuilder = new ();
+        private readonly Dictionary<string, byte[]> files = new ();
+
+        public URLDomain CatalystBaseUrl { get; }
+        public URLDomain ContentBaseUrl { get; }
+        public URLDomain LambdasBaseUrl { get; }
+        public URLDomain EntitiesActiveEndpoint { get; }
+
+        public IReadOnlyList<string> SceneUrns => sceneUrns;
+
+        public IpfsRealm(IWeb3IdentityCache web3IdentityCache,
+            IWebRequestController webRequestController,
+            URLDomain realmName, IpfsTypes.ServerAbout serverAbout = null)
+        {
+            this.web3IdentityCache = web3IdentityCache;
+            this.webRequestController = webRequestController;
+
+            // TODO: realmName resolution, for now just accepts custom realm paths...
+            CatalystBaseUrl = realmName;
+
+            if (serverAbout != null)
+            {
+                sceneUrns = serverAbout.configurations.scenesUrn;
+                ContentBaseUrl = URLDomain.FromString(serverAbout.content.publicUrl);
+                LambdasBaseUrl = URLDomain.FromString(serverAbout.lambdas.publicUrl);
+
+                //Note: Content url requires the subdirectory content, but the actives endpoint requires the base one.
+                EntitiesActiveEndpoint = URLBuilder.Combine(ContentBaseUrl, URLSubdirectory.FromString("entities/active"));
+                ContentBaseUrl = URLBuilder.Combine(ContentBaseUrl, URLSubdirectory.FromString("contents/"));
+            }
+            else
+            {
+                ContentBaseUrl = URLBuilder.Combine(CatalystBaseUrl, URLSubdirectory.FromString("content/contents/"));
+                EntitiesActiveEndpoint = URLBuilder.Combine(CatalystBaseUrl, URLSubdirectory.FromString("content/entities/active"));
+            }
+        }
+
+        public bool Equals(IpfsRealm other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return CatalystBaseUrl == other.CatalystBaseUrl;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+            return Equals((IpfsRealm)obj);
+        }
+
+        public override int GetHashCode() =>
+            ContentBaseUrl.GetHashCode();
+
+        public async UniTask PublishAsync<T>(IpfsRealmEntity<T> entity, IReadOnlyDictionary<string, byte[]> contentFiles, CancellationToken ct)
```

Comment:
I'd allow `contentFiles` be `null` as we will not be using them in the future anyway